internal/cmd/devices/list.go

```diff
@@ -0,0 +1,272 @@
+package devices
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/devices"
+	"github.com/saucelabs/saucectl/internal/devices/devicestatus"
+	"github.com/saucelabs/saucectl/internal/tables"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	JSONOutput = "json"
+	TextOutput = "text"
+)
+
+type DeviceWithStatus struct {
+	ID     string `json:"id"`
+	Name   string `json:"name"`
+	OS     string `json:"os"`
+	Status string `json:"status"`
+}
+
+type Filter struct {
+	Name   string
+	Os     string
+	Status string
+}
+
+type ListOptions struct {
+	Page         int
+	Size         int
+	Status       bool
+	OutputFormat string
+	Filter       Filter
+}
+
+func ListCommand() *cobra.Command {
+	var out string
+	var page int
+	var size int
+	var nameFilter string
+	var osFilter string
+	var addStatus bool
+	var statusFilter string
+
+	cmd := &cobra.Command{
+		Use: "list",
+		Aliases: []string{
+			"ls",
+		},
+		Short:        "Returns the list of devices",
+		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, _ []string) {
+			tracker := usage.DefaultClient
+
+			go func() {
+				tracker.Collect(
+					cmds.FullName(cmd),
+					usage.Flags(cmd.Flags()),
+				)
+				_ = tracker.Close()
+			}()
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if page < 0 {
+				return errors.New("invalid page")
+			}
+			if size < 0 {
+				return errors.New("invalid size")
+			}
+			if out != JSONOutput && out != TextOutput {
+				return errors.New("unknown output format")
+			}
+
+			if statusFilter != "" {
```

Comment:
Shouldn't the status be validated here too?

Reply:
good idea, added